### PR TITLE
Add 'address build-*' commands

### PR DESCRIPTION
**Pull request by Jimbo4350**
Issue
-----------

- This PR **results** in breaking changes to upstream dependencies.

Checklist
---------
- [ ] This PR contains all the work required to resolve the linked issue.

- [ ] The work contained has sufficient documentation to describe what it does and how to do it.

- [ ] The work has sufficient tests and/or testing.

- [ ] I have committed clear and descriptive commits. Be considerate as somebody else will have to read these.

- [ ] I have added the appropriate labels to this PR.

### CODE REVIEW

**Comment by Jimbo4350** [comment]
Once the names are decided upon I can add a test for the new command

File: cardano-api/src/Cardano/Api.hs [hunk 1]
```diff
@@ -263,6 +266,18 @@ shelleyVerificationKeyAddress vkey _nw =
         Shelley.Addr (Shelley.KeyHashObj (Shelley.hashKey vk))
                      Shelley.StakeRefNull
 
+-- | Builds a shelley base address. This type of address has staking rights and
+-- therefore requires a payment verification key and a staking verification key.
+shelleyVerificationKeyBaseAddress :: ShelleyVerificationKey -> ShelleyVerificationKeyStaking -> ShelleyAddress
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api.hs:271]: I think I'd stick to one name, e.g. `shelleyVerificationKeyAddress` but stick a `Maybe` on the `ShelleyVerificationKeyStaking`.

File: cardano-api/src/Cardano/Api/Types.hs [hunk 2]
```diff
@@ -128,7 +129,7 @@ type ShelleyCertificate                  = Shelley.DCert Shelley.TPraosStandardC
 type ShelleyCoin                         = Shelley.Coin
 type ShelleyCredentialStaking            = Shelley.Credential Shelley.Staking Shelley.TPraosStandardCrypto
 type ShelleyCredentialStakePool          = Shelley.Credential Shelley.StakePool Shelley.TPraosStandardCrypto
-type ShelleyVerificationKey              = Shelley.VKey Shelley.Payment Shelley.TPraosStandardCrypto
+type ShelleyVerificationKeyPayment       = Shelley.VKey Shelley.Payment Shelley.TPraosStandardCrypto
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Types.hs:132]: :+1:

File: cardano-cli/src/Cardano/CLI/Shelley/Run.hs [hunk 3]
```diff
@@ -277,11 +282,19 @@ runStakePoolRegistrationCert
...
 
     -- Pool reward account
-    rewardAcctVerKey <- firstExceptT CardanoApiError . newExceptT $ readVerificationKeyStaking rwdVerFp
+    StakingVerificationKeyShelley rewardAcctVerKey <-
+      firstExceptT CardanoApiError . newExceptT $ readStakingVerificationKey rwdVerFp
     let rewardAccount = Shelley.mkRwdAcnt . mkShelleyStakingCredential $ hashKey rewardAcctVerKey
 
     -- Pool owner(s)
-    sPoolOwnerVkeys <-  mapM (\(VerificationKeyFile fp) -> firstExceptT CardanoApiError $ newExceptT $ readVerificationKeyStaking fp) ownerVerFps
+    sPoolOwnerVkeys <-
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run.hs:290]: :+1:

**Comment by Jimbo4350** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**